backend/pkg/logger/data/logger.go

```diff
@@ -0,0 +1,135 @@
+package data
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/abstraction"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/logger"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/transport/packet/data"
+)
+
+const (
+	Name abstraction.LoggerName = "data"
+)
+
+// Logger is a struct that implements the abstraction.Logger interface
+type Logger struct {
+	// An atomic boolean is used in order to use CompareAndSwap in the Start and Stop methods
+	running     *atomic.Bool
+	runningLock sync.RWMutex
+	// initialTime fixes the starting time of the log
+	initialTime time.Time
+	// valueFileSlice is a map that contains the file of each value
+	valueFileSlice map[data.ValueName]io.WriteCloser
+}
+
+// Record is a struct that implements the abstraction.LoggerRecord interface
+type Record struct {
+	packet *data.Packet
+}
+
+func (data *Record) Name() abstraction.LoggerName {
+	return Name
+}
+
+func (sublogger *Logger) Start() error {
+	if !sublogger.running.CompareAndSwap(false, true) {
+		fmt.Println("Logger already running")
+		return nil
+	}
+	sublogger.initialTime = time.Now()
+
+	fmt.Println("Logger started")
+	return nil
+}
+
+// numeric is an interface that allows to get the value of any numeric format
+type numeric interface {
+	Value() float64
+}
+
+func (sublogger *Logger) PushRecord(record abstraction.LoggerRecord) error {
+	if !sublogger.running.Load() {
+		return &logger.ErrLoggerNotRunning{
+			Name:      Name,
+			Timestamp: time.Now(),
+		}
+	}
+
+	if reflect.TypeOf(record) != reflect.TypeOf(&Record{}) {
+		return &logger.ErrWrongRecordType{
+			Name:      Name,
+			Timestamp: time.Now(),
+			Expected:  &Record{},
+			Received:  record,
+		}
+	}
```

Comment:
It is usually possible to do anything without using reflect, just use reflect when strictly necessary because it makes the code less readable
```suggestion
	dataRecord, ok := record.(*Record)
	if !ok {
	    return &logger.ErrWrongRecordType{
	        //...
	    }
	}
	
	// use data record here...
```